### PR TITLE
Handle non-JSON responses in callApi() correctly

### PR DESCRIPTION
**Pull request by kumar303**
Fixes https://github.com/mozilla/addons-frontend/issues/1701

### CODE REVIEW

File: src/core/api/index.js [hunk 1]
```diff
@@ -60,15 +61,18 @@ export function callApi({
...
-      (jsonResponse) => ({ response, jsonResponse }),
-      (error) => {
-        log.warn('Could not parse response as JSON:', error);
-        return response.text().then((textResponse) =>
-          ({ response, jsonResponse: { text: textResponse } })
+    .then((response) => {
+      return response.clone().json()
+        .then(
+          (jsonResponse) => ({ response, jsonResponse }),
+          (error) => {
```
Comment (tofumatt) [inline src/core/api/index.js]: I feel like I've asked this before for `fetch` stuff but I forget: any reason to use the much-less-seen error argument over `.catch`?

File: src/core/api/index.js [hunk 2]
```diff
@@ -1,4 +1,5 @@
 /* global fetch */
+/* eslint-disable arrow-body-style */
```
Comment (tofumatt) [inline src/core/api/index.js]: I'm guessing this is for https://github.com/mozilla/addons-frontend/pull/1702/files#diff-9c6af2bc9310bfed1e429d02f827c559R65? If so, maybe this should just be a `disable-next-line`?

It's unrelated but I'd love to switch over to `arrow-body-style` always; I find it so much easier to scan.

Comment (kumar303) [inline src/core/api/index.js]: You're going to fight me for it! It's about to die https://github.com/mozilla/addons-frontend/issues/1252 I generally agree that single value returns should always use arrow body but long promise chains get really hard to read. In fact, when I first submitted the code for review, mark had trouble with it because it was hard to follow the `response` variable reference.

File: src/core/api/index.js (hunk 1 above)
Comment (kumar303) [inline src/core/api/index.js]: The same code won't work with catch. If I put a catch above this and recover, it will execute the `jsonResponse` then function. I don't think there's another way to do it.

These kind of patterns is why promise chains are terrible! If we move to async/await (https://github.com/mozilla/addons-frontend/issues/1473) everything will be way easier to read.

File: src/core/api/index.js (hunk 2 above)
Comment (tofumatt) [inline src/core/api/index.js]: Oh maybe I was unclear, I actually would prefer we used explicit `return` *always*; I dislike the implicit return syntax and invariable refactoring of the code once it becomes multi-line/statement.

Anywho, this is better, I was just nitpicking about disabling a rule whole-file over line, really 😄

File: src/core/api/index.js (hunk 1 above)
Comment (tofumatt) [inline src/core/api/index.js]: Ahh, gotcha. Yeah the fetch API is a bit odd. I need to learn the async/await syntax, though it def looks nicer 😄

**Comment by tofumatt** [comment]
I think these test failures are just `servertest` related but I'm not sure. r+ to the code but be sure it passes tests; I re-ran them a few times last night and they're still failing...

File: src/core/api/index.js (hunk 2 above)
Comment (kumar303) [inline src/core/api/index.js]: oh, I see what you mean. Well, I like doing things like `someArray.filter((value) => value === true)`. I wouldn't want to give that up.
